File: TEMA 6/ListaMultimedia/src/model/Persona.java
package model;

public class Persona {
    private String nombre;
    private String dni;

    public Persona(){}

    public Persona(String nombre, String dni){
        this.nombre = nombre;
        this.dni = dni;
    }

    public void mostrarDatos(){
        System.out.println("El nombre es "+nombre);
        System.out.println("El dni es "+dni);
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getDni() {
        return dni;
    }

    public void setDni(String dni) {
        this.dni = dni;
    }
}

File: DAM2/PMDM/TrabajadorHerencia/src/model/Persona.kt
package model

abstract class Persona (var nombre : String, var apellido : String, var DNI : String) {
    //nombre, apellido, DNI-> obligatorio
    // telefono, correo ->opcional

    var telefono : Int? = null

    var correo : String? = null

    constructor(nombre: String,apellido: String,DNI: String, telefono: Int, correo: String):this(
        nombre, apellido, DNI){
        this.telefono = telefono
        this.correo = correo
    }

    open fun mostrarDatos(){
        println("Nombre: $nombre")
        println("Apellido: $apellido")
        println("DNI: $DNI")
        println("El telefono es $telefono?: no se ha especificado")
        println("El correo es $correo?: no se ha especificado")
    }
}